conda/gateways/disk/read.py

```diff
@@ -45,17 +46,44 @@ def collect_all_info_for_package(extracted_package_directory):
     info_dir = join(extracted_package_directory, 'info')
 
     files_path = join(extracted_package_directory, 'info', 'files')
-    files = tuple(ln for ln in (line.strip() for line in yield_lines(files_path)) if ln)
+    file_json_path = join(extracted_package_directory, 'info', 'files.json')
+
+    if isfile(file_json_path):
+        with open(file_json_path) as file_json:
+            data = json.load(file_json)
+        if data.get('version') != 1:
+            raise CondaUpgradeError("Expected files.json schema to be version 1")
```

Comment:
"The current version of conda is too old to install this package. (This version only supports files.json schema version 1.)  Please update conda to install this package."